### ISSUE TITLE
Error: Clerk: auth() was called but it looks like you aren't using `authMiddleware` in your middleware file.

### ISSUE THREAD

**Issue by krutikbundela** [adrianhajdin/threads#77]
Unhandled Runtime Error
Error: Clerk: auth() was called but it looks like you aren't using `authMiddleware` in your middleware file. Please use `authMiddleware` and make sure your middleware matcher is configured correctly and it matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs

The middleware.ts is on same level as app
![image](https://github.com/adrianhajdin/threads/assets/116296880/3e71c021-f739-4f3c-be68-cc8eef9a58a8)

not able to figureout what is the issue if someone can help tha would be really greatfull

**Comment by NammDev** [comment]
Did you find solution for this?

**Comment by MichelHMachado** [comment]
Same problem here.

**Comment by Toristeju112** [comment]
> Same problem here. Changed to node 20.1.0 and worked.

Yes, This worked, noticed you have to use the right npm package when working with clerk, some are not compatible with the latest version using version node 18.0.0 will initialize your nextjs but won't work with the api routing and authmiddleware, but when tested on new version (20.1.0) it worked.
- nvm install 20.1.0
## if you don't want to add it permanently to your bash after closing your terminal.
you can use it again with (nvm use 20.1.0) when working on a new terminal to prevent the authmiddleware error.

**Comment by Adedayo-Onabamiro** [comment]
i installed (upgraded) node by downloading it from their site, then re ran my project, but it does'nt still work, can anyone provide a more detailed way to resolve this issue? thanks!

**Comment by Toristeju112** [comment]
> i installed (upgraded) node by downloading it from their site, then re ran my project, but it does'nt still work, can anyone provide a more detailed way to resolve this issue? thanks!

make sure your middleware.ts is at the root directory, then if you are using clerk@latest install nvm i 20.1.0.
confirm by checking the node -v ,  it must be node v - 20.1.0 .
check middleware configuration and  re run

**Comment by Adedayo-Onabamiro** [comment]
honestly tried these and still isnt working, have you completed the project? so i could get like a starter file and all ill have to do is npm i

**Comment by Toristeju112** [comment]
> honestly tried these and still isnt working, have you completed the project? so i could get like a starter file and all ill have to do is npm i

which project exactly? 
what's your node version, nextjs version and clerk version

**Comment by Adedayo-Onabamiro** [comment]
the threads project. my node is 20.3.1, my next is 13.5.4 and my clerk is 4.25.2
i used npm and npx to check all these versions, although currently, i copied JSM's package.json, to try to fix my error and did all sorts and it didnt work.

**Comment by Toristeju112** [comment]
> the threads project. my node is 20.3.1, my next is 13.5.4 and my clerk is 4.25.2 i used npm and npx to check all these versions, although currently, i copied JSM's package.json, to try to fix my error and did all sorts and it didnt work.

 the threads project, okay 
use:
Next: 13.4.19
Node: 20.1.0
clerk: 4.25.1

is a more stable version.

**Comment by Digitl-Alchemyst** [comment]
> the threads project. my node is 20.3.1, my next is 13.5.4 and my clerk is 4.25.2 i used npm and npx to check all these versions, although currently, i copied JSM's package.json, to try to fix my error and did all sorts and it didnt work.



> > the threads project. my node is 20.3.1, my next is 13.5.4 and my clerk is 4.25.2 i used npm and npx to check all these versions, although currently, i copied JSM's package.json, to try to fix my error and did all sorts and it didnt work.
> 
> the threads project, okay use: Next: 13.4.19 Node: 20.1.0 clerk: 4.25.1
> 
> is a more stable version.

I am running this project as well and here is the versions I am using
"@clerk/nextjs": "^4.25.3",
"next": "13.5.4",
Node: v20.8.0

I was getting the error trying several version combos on this one, my problem was I am using a source folder and had the middleware file in the root dir, once I moved it to the src folder everything worked as expected.

**Comment by luketas** [comment]
I'm following Antonio's AI companion tutorial and this got me completely stuck..
Tried all the node versions, no idea what will solve this. Anyone had any luck on this particular project? middleware ts is on the same level as the app folder, node v is 20.1.0.. everything seems right. 

![image](https://github.com/adrianhajdin/threads/assets/11461171/0fea98ac-8e36-422e-8f35-b33d5bfac0a7)
![image](https://github.com/adrianhajdin/threads/assets/11461171/48cbddec-c3ca-4f93-a8cc-e70797c03846)

**Comment by AniketBajaj2024** [comment]
> I'm following Antonio's AI companion tutorial and this got me completely stuck.. Tried all the node versions, no idea what will solve this. Anyone had any luck on this particular project? middleware ts is on the same level as the app folder, node v is 20.1.0.. everything seems right.
> 
> ![image](https://private-user-images.githubusercontent.com/11461171/277130054-0fea98ac-8e36-422e-8f35-b33d5bfac0a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4NzEyNTksIm5iZiI6MTcwMzg3MDk1OSwicGF0aCI6Ii8xMTQ2MTE3MS8yNzcxMzAwNTQtMGZlYTk4YWMtOGUzNi00MjJlLThmMzUtYjMzZDViZmFjMGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI5VDE3MjkxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYyZGM0MmU2NmQ3MmI0MGJlNjUwOGMzMDU3OGRlNzlhYTNhOTRhYjExNWRkNGQxNjI0MDUzM2QxYWM1YzJhN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.izU3yoEHDXjCissxLHq0bh_gtOyTJftJl9xyKrt28-A) ![image](https://private-user-images.githubusercontent.com/11461171/277130076-48cbddec-c3ca-4f93-a8cc-e70797c03846.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4NzEyNTksIm5iZiI6MTcwMzg3MDk1OSwicGF0aCI6Ii8xMTQ2MTE3MS8yNzcxMzAwNzYtNDhjYmRkZWMtYzNjYS00ZjkzLWE4Y2MtZTcwNzk3YzAzODQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI5VDE3MjkxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwY2YzZGRmYjIzZjMxMTU2NjBkODA4MGEwYmM0OTZkOWJkNWI2OTM4ZWI2OWU1YzQ3MjBhY2JkZjBhMzU2MTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CGwIempC8iDH1resXbvprC3ZmjxdhHFx_9x9yKeSnds)

bro have u got the solution?

**Comment by Stackustack** [comment]
Im stuck on:
> node: 20.11.0
> clerk: 4.29.9
> next: 14.1.3

This is really frustrating. As for now I just disabled authentication by removing the middleware part from `/api/uploadthing/core.ts` as `onUploadComplete` doesn't do much with user data anyway. Obviously this is not production ready since it doesn't authenticate the user when uploading stuff... but whatever, its just educational course 👍 

This is what you get when you create tutorials with `xxx@latests` 😮‍💨 if it works at time when @adrianhajdin creates it, it might not work on latest packages 🤷
